backend/dataall/modules/shares_base/services/share_notification_service.py

```diff
@@ -56,11 +56,50 @@ def notify_share_object_submission(self, email_id: str):
         self._create_notification_task(subject=subject, msg=email_notification_msg)
         return notifications
 
+    def notify_persistent_email_reminder(self, email_id: str, engine):
+        share_link_text = ''
+        if os.environ.get('frontend_domain_url'):
```

Comment:
ShareNotificationService is a generic service that can be used in APIs/Worker... We should not assume that `os.environ.get('frontend_domain_url'):` is part of the env_vars. Maybe we should pass it as a function variable

Reply:
Hmm, we have similar code in `notify_share_object_submission` function. 
```
    def notify_share_object_submission(self, email_id: str):
        share_link_text = ''
        if os.environ.get('frontend_domain_url'):
            share_link_text = f'<br><br> Please visit data.all <a href="{os.environ.get("frontend_domain_url")}/console/shares/{self.share.shareUri}">share link </a> to take action or view more details'
        msg = f'User {email_id} SUBMITTED share request for dataset {self.dataset.label} for principal {self.share.principalId}'
        subject = f'Data.all | Share Request Submitted for {self.dataset.label}'
        email_notification_msg = msg + share_link_text

        notifications = self.register_notifications(
            notification_type=DataSharingNotificationType.SHARE_OBJECT_SUBMITTED.value, msg=msg
        )

        self._create_notification_task(subject=subject, msg=email_notification_msg)
        return notifications
```

Hence, I thought it might be okay to re-use it. LMK if that doesnt work and what might be a better way to address it.

Reply:
Looking at the rest of the code, if it is not present the message will still be sent (just without the redirect link). So it is not catastrophical that it is not passed. We can leave it as it is for now